prompts/template_xml/context.txt

```diff
@@ -1,8 +1,30 @@
-The function is defined in
-<include_file>
-{CONTEXT_HEADER}
-</include_file>
+{% if headers %}
 
+The code block below contains the header files which you must include. The function requires these header files to successfully compile.
+If the headers are private, do not include them.
+Header files:
 <code>
-{CONTEXT_TYPES}
+{{ headers }}
 </code>
+{% endif %}
+{% if must_insert %}
+
+You must insert code in the below code block before the function being tested:
+<code>
+{{ must_insert }}
+</code>
+{% endif %}
+{% if func_source %}
+
+Here is the source code of the function being tested:
+<code>
+{{ func_source }}
+</code>
+{% endif %}
+{% if xrefs %}
+
+Here is the source code for functions which reference the function being tested:
+<code>
+{{ xrefs }}
```

Comment:
When this works, it seems like we are including the entire source code of the functions that call the function-to-fuzz. e.g. on https://llm-exp.oss-fuzz.com/Result-reports/ofg-pr/2024-05-03-192-test-244-comparison/benchmark/output-libvpx-parse_block/index.html 

This is **a lot** of context. Would there be an easy way to instead say, include up to 10 lines before and 10 lines after the relevant function references? 

I.e.

```
... 10 previous lines ...
<function call>
... 10 lines after ...
```

Rather than the entire source code?

@DavidKorczynski for thoughts too.

Reply:
i could the split returned code into lines and then do a string match and truncate based on that, though there might be complications with c++.